msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase1561125.java

```diff
@@ -0,0 +1,133 @@
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+
+// [Joined][MSAL] In-line WPJ: Perform Device registration with deviceid claim
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1561125
+public class TestCase1561125 extends AbstractMsalBrokerTest {
+    @Test
+    public void test_1561125() throws Throwable {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        final MsalSdk msalSdk = new MsalSdk();
+        // create claims request object
+        final ClaimsRequest claimsRequest = new ClaimsRequest();
+        final RequestedClaimAdditionalInformation requestedClaimAdditionalInformation =
+                new RequestedClaimAdditionalInformation();
+
+        requestedClaimAdditionalInformation.setEssential(true);
+
+        // request the deviceid claim in ID Token
+        claimsRequest.requestClaimInIdToken("deviceid", requestedClaimAdditionalInformation);
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.LOGIN)
+                .claims(claimsRequest)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+
+        // start interactive acquire token request in MSAL (should succeed)
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new com.microsoft.identity.client.ui.automation.interaction.OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.LOGIN)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .registerPageExpected(true)
+                        .speedBumpExpected(false)
+                        .broker(mBroker)
+                        .expectingLoginPageAccountPicker(false)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+        Map<String, ?> tokens = IDToken.parseJWT(authResult.getAccessToken());
+        if (tokens.containsKey("deviceid")) {
+            Assert.assertNotNull(tokens.get("deviceid"));
+        }
+        else {
+            Assert.fail("decoded AccessToken does not contain the deviceId");
+        }
```

Comment:
It can be simplified to the following

```suggestion
        Assert.assertNotNull(tokens.get("deviceid"));
```